### PR TITLE
AP-1152 Dependants collator

### PR DESCRIPTION
**Pull request by stephenrichards**
## Calculate and sum dependants allowances

[Link to story](https://dsdmoj.atlassian.net/browse/AP-1152)

- Added `dependant_allowance` field to `Dependant`
- Added `monthly_dependants_allowance` field to `DisposableIncomeSummary`
- Created `Calculators::DependantsAllowanceCalculator` to determine allowance for each dependant
- Created `Collators::DependantAllowanceCollator` to call the calculator for each dependant and update the dependant records with the result, and write the sum of the results to the `DisposableIncomeSummary` record.

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/services/calculators/dependant_allowance_calculator.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    def call
+      return child_under_15_allowance if under_15_years_old?
+
+      return child_aged_15_allowance - monthly_income if under_16_years_old?
+
+      return child_16_and_over_allowance - monthly_income if under_18_in_full_time_education?
+
+      return 0.0 if capital_over_allowance?
+
+      adult_allowance - @dependant.monthly_income
```
Comment (colinbruce) [inline app/services/calculators/dependant_allowance_calculator.rb]: `@dependant.monthly_income` here, but `monthly_income` in the two children blocks above

File: app/services/calculators/dependant_allowance_calculator.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    end
+
+    def initialize(dependant)
+      @dependant = dependant
+    end
+
+    def call
+      return child_under_15_allowance if under_15_years_old?
+
+      return child_aged_15_allowance - monthly_income if under_16_years_old?
```
Comment (colinbruce) [inline app/services/calculators/dependant_allowance_calculator.rb]: Could this be extracted to a`child_under_16_allowance` to make this easier to read?

File: app/services/calculators/dependant_allowance_calculator.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+      return child_16_and_over_allowance - monthly_income if under_18_in_full_time_education?
```
Comment (colinbruce) [inline app/services/calculators/dependant_allowance_calculator.rb]: ditto `16_to_18_in_full_time_education_allowance`?

File: spec/services/calculators/dependant_allowance_calculator_spec.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      end
+
+      context '16 or 17 years old in full time education with no income' do
+        let(:dependant) { create :dependant, :aged_16_or_17, monthly_income: 0.0, in_full_time_education: true }
+        it 'returns the child 16 or over with no income deduction' do
+          expect(subject).to eq 333.33
+        end
+      end
+
+      context '16 years old in full time education with income' do
```
Comment (colinbruce) [inline spec/services/calculators/dependant_allowance_calculator_spec.rb]: Should we test that `full_time_education = false` returns `adult_allowance - @dependant.monthly_income`?

File: app/services/calculators/dependant_allowance_calculator.rb (hunk 1 above)
Comment (stephenrichards) [reply app/services/calculators/dependant_allowance_calculator.rb]: Good spot.  Should both be monthly_income

**Pull request merged**